src/modules/logger/logger.cpp

```diff
@@ -701,9 +676,38 @@ void Logger::run()
 				/* if this topic has been updated, copy the new data into the message buffer
 				 * and write a message to the log
 				 */
-				const bool try_to_subscribe = (sub_idx == next_subscribe_topic_index);
+				bool msg_copied = false;
+				void *buffer = _msg_buffer + sizeof(ulog_message_data_header_s);
+
+				if (sub.valid()) {
```

Comment:
Can you keep this in a separate method? It's IMO more readable as the loop is quite long already.

Reply:
Done.